### ISSUE TITLE
TypeError: cannot pickle 'coroutine' object when using async def as default argument in get_or_set

### ISSUE THREAD

**Issue by emielmolenaar** [Krukov/cashews#266]
I have a class which fetches data from an API and database by using async methods. I would like to cache consequent calls to this method:

```
class SomeClass:
  async def get_data(argument):
    data_to_return = await cache.get_or_set(f"cachekey:{argument}", default=self.__get_data_to_return(argument))
    return data_to_return

  async def __get_data_to_return(argument):
    return await get_data_from_api()
```

This raises a "TypeError: cannot pickle 'coroutine' object". What am I doing wrong?

**Comment by Krukov** [comment]
Hello, thanks for reporting this issue

For some reason `get_or_set` do **not** support a coroutine as default. It accepts callable or awaitable objects. 
So I guess that next example will work as you expect 
```python
class SomeClass:
  async def get_data(self, argument):
    async def default():
      return await self.__get_data_to_return(argument)
    data_to_return = await cache.get_or_set(f"cachekey:{argument}", default=default)
    return data_to_return

  async def __get_data_to_return(self, argument):
    return await get_data_from_api()

```

I'm sure it's possible to add support for coroutine objects too.

**Comment by emielmolenaar** [comment]
Thanks for your prompt response! I have been trying to figure out why this happens. I have tried your example, but I still get the same error.

After some investigation, I think the error occurs because I pass an argument to the `__get_data_to_return` method. Because of that, `if callable(default)` (https://github.com/Krukov/cashews/blob/master/cashews/wrapper/commands.py#L52) returns False and the given coroutine isn't awaited before the response is stored in the cache. 

I have modified the `get_or_set` method locally to this:

```
    async def get_or_set(
        self, key: Key, default: Default | AsyncCallable_T | Callable_T, expire: TTL = None, **kwargs
    ) -> Value | Default | Result_T:
        value = await self.get(key, default=_empty)
        if value is not _empty:
            return value
        if callable(default):
            if inspect.iscoroutinefunction(default):
                _default = await default(kwargs['args'])
            else:
                _default = default()
        else:
            _default = default
        await self.set(key, _default, expire=expire)
        return _default
```

1. Added the `**kwargs` argument to the `get_or_set`
2. Used `**kwargs` to pass arguments to the `default` coroutine
3. Called `get_or_set` like this:

```
data_to_return = await cache.get_or_set(f"cachekey:{argument}", default=self.__get_data_to_return, args=argument)
```

That works 👍🏻  However, I am not a Python programmer. So please by advised that this solution might not be the best 😃 Any suggestions welcome, I am willing to write a PR for this.

**Comment by Krukov** [comment]
Hi, sorry for not responding for so long. 

To be honest, it's strange that my example doesn't work and gives the same error, maybe I or you miss something ) 

Any way it will be better to add support for awaitable objects (like coroutine) - without passing arguments via  kwargs 

 For me, it will better to add something like this inside `get_or_set` method:

```python
        if inspect.isawaitable(default):
            _default = await default
        elif callable(default):
            ...

```

**Comment by Krukov** [comment]
released with 7.4.0
